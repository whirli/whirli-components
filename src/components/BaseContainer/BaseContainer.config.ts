// @ts-ignore
import { ComponentStyles, ComponentProps } from '@whirli-components/@types/components';

// Local config
import config from '@whirli-local/components/BaseContainer/BaseContainer.config';

export const ConfigStyles: ComponentStyles = {
  ...config.styles,
};

export const ConfigProps: ComponentProps = {
  ...config.props,
};
