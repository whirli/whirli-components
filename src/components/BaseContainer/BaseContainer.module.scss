@import '../../styles/Breakpoints.scss';

// default styles
.container {
    width: 100%;
    max-width: 64rem;
    padding: 1rem;
    margin: auto;
    box-sizing: border-box;
}

// breakpoint styles
$packageBreakpointStyles: ();