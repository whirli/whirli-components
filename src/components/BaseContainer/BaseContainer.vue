<template>
  <div :class="classes">
    <slot />
  </div>
</template>

<script setup="props" lang="ts">
// Vue
import { defineProps } from 'vue';

// Styles
import styles from '@whirli-local/components/BaseContainer/BaseContainer.module.scss';

const classes = [styles.container];
</script>
