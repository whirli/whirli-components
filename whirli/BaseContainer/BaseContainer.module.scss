@import '@whirli-components/src/components/BaseContainer/BaseContainer.module.scss';

// Standard styles

// Breakpoint styles
$localBreakpointStyles: ();

@include createBreakpointClasses(map-merge($packageBreakpointStyles, $localBreakpointStyles));