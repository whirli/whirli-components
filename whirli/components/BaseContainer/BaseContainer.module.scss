@import '@whirli-components/components/BaseContainer/BaseContainer.module.scss';

// Standard styles

// Breakpoint styles
$localBreakpointStyles: ();

@include create-breakpoint-classes(map-merge($package-breakpoint-styles, $localBreakpointStyles));