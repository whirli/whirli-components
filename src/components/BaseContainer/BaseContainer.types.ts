import { PropKeys, PropValues } from './BaseContainer.constants';

// const propColor = PropValues[PropKeys.COLOR];

// export type PropColor = typeof propColor[number];
