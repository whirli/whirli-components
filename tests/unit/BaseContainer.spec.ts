import { shallowMount } from '@vue/test-utils';
import BaseContainer from '@/components/BaseContainer/BaseContainer.vue';

// eslint-disable-next-line @typescript-eslint/no-explicit-any
let wrapper: any;
const text = 'Hello World';

describe('BaseContainer.vue - Single classes', () => {
  beforeEach(() => {
    wrapper = shallowMount(BaseContainer, {
      slots: {
        default: text,
      },
    });
  });

  it('renders the component', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it('displays the correct text', () => {
    expect(wrapper.text()).toMatch(text);
  });
});
