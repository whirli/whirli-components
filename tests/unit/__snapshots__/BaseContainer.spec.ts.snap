// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseContainer.vue - Single classes renders the component 1`] = `
<div
  class=""
>
  
  Hello World
  
</div>
`;
